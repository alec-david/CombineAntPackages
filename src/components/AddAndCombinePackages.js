import React, { Component } from 'react';
import Dropzone from 'react-dropzone';
import Clipboard from 'clipboard';
import Textarea from 'react-textarea-autosize';

import * as styles from '../utils/styles';
import { combineSelectedPackages } from '../utils/CombinePackagesLogic';

new Clipboard('.btn');

class AddPackage extends Component {
  constructor() {
    super();
    this.state = {
      files: [],
      packageXML: ''
    };
  }

  onDrop(files) {
    if (files.length > 0) {
      let combinePackageButton = this.combineButton;
      combinePackageButton.style.display = 'inline';
    }
    this.setState({
      files
    });
  }

  combinePackages() {
    let combinePromise = combineSelectedPackages(this.state.files);
    combinePromise.then(result => {
      this.setState({
        packageXML: result
      });
    });
  }

  fileDownload(data, filename, mime) {
    let blob = new Blob([data], { type: mime || 'application/octet-stream' });

    // IE workaround
    if (typeof window.navigator.msSaveBlob !== 'undefined') {
      window.navigator.msSaveBlob(blob, filename);
    } else {
      let blobURL = window.URL.createObjectURL(blob);
      let tempLink = document.createElement('a');
      tempLink.style.display = 'none';
      tempLink.href = blobURL;
      tempLink.setAttribute('download', filename);

      // Safari thinks _blank anchor are pop ups. We only want to set _blank
      // target if the browser does not support the HTML5 download attribute.
      if (typeof tempLink.download === 'undefined') {
        tempLink.setAttribute('target', '_blank');
      }

      document.body.appendChild(tempLink);
      tempLink.click();
      document.body.removeChild(tempLink);
      window.URL.revokeObjectURL(blobURL);
    }
  }

  render() {
    let combinedPackage = this.state.packageXML;

    if (!combinedPackage) {
      return (
        <section>
          <div className="dropzone">
            <Dropzone
              onDrop={this.onDrop.bind(this)}
              accept=".xml"
              style={styles.dropzoneStyle}>
              <p>
                Try dropping some files here, or click to select files to
                upload.
              </p>
            </Dropzone>
          </div>
          <div
            id="combineButton"
            ref={buttonDiv => {
              this.combineButton = buttonDiv;
            }}
            style={{ display: 'none' }}>
            <aside>
              <h2>Selected files</h2>
              <div>
                {this.state.files.map(f => (
                  <span key={f.name}>
                    {f.name} - {f.size} bytes
                    <br />
                    <br />
                  </span>
                ))}
              </div>
            </aside>
            <input
              type="button"
              value="Combine Packages"
              onClick={this.combinePackages.bind(this)}
              style={styles.buttonStyle}
            />
          </div>
        </section>
      );
    }
    return (
      <div>
        <button
          className="btn"
          data-clipboard-text={combinedPackage}
          style={styles.buttonStyle}>
          Copy package.xml to clipboard
        </button>
        <button
          onClick={() => this.fileDownload(combinedPackage, 'package.xml')}
          style={styles.buttonStyle}>
          Download Combined Package
        </button>
        <button
          onClick={() => this.setState({ packageXML: '', files: [] })}
          style={styles.buttonStyle}>
          Create New Combined Package
        </button>
        <br />
        <br />
        <br />
        <h3>Combined Package:</h3>
        <br />
        <br />
        <div style={styles.formatPackage}>
          <Textarea
            name="packageTextArea"
            value={combinedPackage}
            style={styles.textAreaStyle}
            spellcheck="false"
          />
        </div>
      </div>
    );
  }
}

export default AddPackage;
