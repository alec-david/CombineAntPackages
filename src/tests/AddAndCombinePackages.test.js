import React from "react";
import ReactDOM from "react-dom";

import { mount } from "enzyme";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import AddAndCombinePackages from "../components/AddAndCombinePackages";

describe("Creating the component does not fail", () => {
  const wrapper = shallow(<AddAndCombinePackages />);

  it("should match its empty snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Calling the components functions shallow render", () => {
  const wrapper = shallow(<AddAndCombinePackages />);

  wrapper.instance().onDrop([]);
  it("should not render button to combine files", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Calling components function", () => {
  const wrapper = mount(<AddAndCombinePackages />);

  wrapper.instance().onDrop([generateFile(1)]);
  wrapper.update();
  it("1 file added -> should render button to combine files", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  wrapper.instance().onDrop([generateFile(1), generateFile(2)]);
  it("2 files added -> should render button to combine files", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  wrapper.instance().onDrop([]);
  it("0 files added -> should not render button to combine files", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

function generateFile(num) {
  const fileObj = {
    lastModified: 1504619161306,
    lastModifiedDate: new Date(1504619161306),
    name: "package" + num + ".xml",
    preview: "asuh",
    size: 123456,
    type: "text/xml",
    webkitRelativePath: "",
    key: num
  };
  return fileObj;
}
