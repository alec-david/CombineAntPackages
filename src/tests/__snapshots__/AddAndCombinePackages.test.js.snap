// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calling components function 0 files added -> should not render button to combine files 1`] = `
<AddPackage>
  <section>
    <div
      className="dropzone"
    >
      <Dropzone
        accept=".xml"
        disableClick={false}
        disablePreview={false}
        disabled={false}
        maxSize={Infinity}
        minSize={0}
        multiple={true}
        onDrop={[Function]}
        preventDropOnDocument={true}
        style={
          Object {
            "borderColor": "#666666",
            "borderRadius": "5px",
            "borderStyle": "dashed",
            "borderWidth": "2px",
            "height": "200px",
            "width": "400px",
          }
        }
      >
        <div
          aria-disabled={false}
          className=""
          onClick={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          style={
            Object {
              "borderColor": "#666666",
              "borderRadius": "5px",
              "borderStyle": "dashed",
              "borderWidth": "2px",
              "height": "200px",
              "width": "400px",
            }
          }
        >
          <p>
            Try dropping some files here, or click to select files to upload.
          </p>
          <input
            accept=".xml"
            autoComplete="off"
            disabled={false}
            multiple={true}
            onChange={[Function]}
            style={
              Object {
                "display": "none",
              }
            }
            type="file"
          />
        </div>
      </Dropzone>
    </div>
    <div
      id="combineButton"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <aside>
        <h2>
          Selected files
        </h2>
        <div />
      </aside>
      <input
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#2F4F4F",
            "border": "1px solid #566963",
            "borderRadius": "5px",
            "color": "#ffffff",
            "cursor": "pointer",
            "display": "inline-block",
            "fontSize": "15px",
            "fontWeight": "bold",
            "height": "50px",
            "margin": "10px",
            "textDecoration": "none",
            "textShadow": "0px-1px0px #2b665e",
          }
        }
        type="button"
        value="Combine Packages"
      />
    </div>
  </section>
</AddPackage>
`;

exports[`Calling components function 1 file added -> should render button to combine files 1`] = `
<AddPackage>
  <section>
    <div
      className="dropzone"
    >
      <Dropzone
        accept=".xml"
        disableClick={false}
        disablePreview={false}
        disabled={false}
        maxSize={Infinity}
        minSize={0}
        multiple={true}
        onDrop={[Function]}
        preventDropOnDocument={true}
        style={
          Object {
            "borderColor": "#666666",
            "borderRadius": "5px",
            "borderStyle": "dashed",
            "borderWidth": "2px",
            "height": "200px",
            "width": "400px",
          }
        }
      >
        <div
          aria-disabled={false}
          className=""
          onClick={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          style={
            Object {
              "borderColor": "#666666",
              "borderRadius": "5px",
              "borderStyle": "dashed",
              "borderWidth": "2px",
              "height": "200px",
              "width": "400px",
            }
          }
        >
          <p>
            Try dropping some files here, or click to select files to upload.
          </p>
          <input
            accept=".xml"
            autoComplete="off"
            disabled={false}
            multiple={true}
            onChange={[Function]}
            style={
              Object {
                "display": "none",
              }
            }
            type="file"
          />
        </div>
      </Dropzone>
    </div>
    <div
      id="combineButton"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <aside>
        <h2>
          Selected files
        </h2>
        <div />
      </aside>
      <input
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#2F4F4F",
            "border": "1px solid #566963",
            "borderRadius": "5px",
            "color": "#ffffff",
            "cursor": "pointer",
            "display": "inline-block",
            "fontSize": "15px",
            "fontWeight": "bold",
            "height": "50px",
            "margin": "10px",
            "textDecoration": "none",
            "textShadow": "0px-1px0px #2b665e",
          }
        }
        type="button"
        value="Combine Packages"
      />
    </div>
  </section>
</AddPackage>
`;

exports[`Calling components function 2 files added -> should render button to combine files 1`] = `
<AddPackage>
  <section>
    <div
      className="dropzone"
    >
      <Dropzone
        accept=".xml"
        disableClick={false}
        disablePreview={false}
        disabled={false}
        maxSize={Infinity}
        minSize={0}
        multiple={true}
        onDrop={[Function]}
        preventDropOnDocument={true}
        style={
          Object {
            "borderColor": "#666666",
            "borderRadius": "5px",
            "borderStyle": "dashed",
            "borderWidth": "2px",
            "height": "200px",
            "width": "400px",
          }
        }
      >
        <div
          aria-disabled={false}
          className=""
          onClick={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          style={
            Object {
              "borderColor": "#666666",
              "borderRadius": "5px",
              "borderStyle": "dashed",
              "borderWidth": "2px",
              "height": "200px",
              "width": "400px",
            }
          }
        >
          <p>
            Try dropping some files here, or click to select files to upload.
          </p>
          <input
            accept=".xml"
            autoComplete="off"
            disabled={false}
            multiple={true}
            onChange={[Function]}
            style={
              Object {
                "display": "none",
              }
            }
            type="file"
          />
        </div>
      </Dropzone>
    </div>
    <div
      id="combineButton"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <aside>
        <h2>
          Selected files
        </h2>
        <div />
      </aside>
      <input
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#2F4F4F",
            "border": "1px solid #566963",
            "borderRadius": "5px",
            "color": "#ffffff",
            "cursor": "pointer",
            "display": "inline-block",
            "fontSize": "15px",
            "fontWeight": "bold",
            "height": "50px",
            "margin": "10px",
            "textDecoration": "none",
            "textShadow": "0px-1px0px #2b665e",
          }
        }
        type="button"
        value="Combine Packages"
      />
    </div>
  </section>
</AddPackage>
`;

exports[`Calling the components functions shallow render should not render button to combine files 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <section>
    <div
      className="dropzone"
    >
      <Dropzone
        accept=".xml"
        disableClick={false}
        disablePreview={false}
        disabled={false}
        maxSize={Infinity}
        minSize={0}
        multiple={true}
        onDrop={[Function]}
        preventDropOnDocument={true}
        style={
          Object {
            "borderColor": "#666666",
            "borderRadius": "5px",
            "borderStyle": "dashed",
            "borderWidth": "2px",
            "height": "200px",
            "width": "400px",
          }
        }
      >
        <p>
          Try dropping some files here, or click to select files to upload.
        </p>
      </Dropzone>
    </div>
    <div
      id="combineButton"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <aside>
        <h2>
          Selected files
        </h2>
        <div />
      </aside>
      <input
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#2F4F4F",
            "border": "1px solid #566963",
            "borderRadius": "5px",
            "color": "#ffffff",
            "cursor": "pointer",
            "display": "inline-block",
            "fontSize": "15px",
            "fontWeight": "bold",
            "height": "50px",
            "margin": "10px",
            "textDecoration": "none",
            "textShadow": "0px-1px0px #2b665e",
          }
        }
        type="button"
        value="Combine Packages"
      />
    </div>
  </section>,
  "nodes": Array [
    <section>
      <div
        className="dropzone"
      >
        <Dropzone
          accept=".xml"
          disableClick={false}
          disablePreview={false}
          disabled={false}
          maxSize={Infinity}
          minSize={0}
          multiple={true}
          onDrop={[Function]}
          preventDropOnDocument={true}
          style={
            Object {
              "borderColor": "#666666",
              "borderRadius": "5px",
              "borderStyle": "dashed",
              "borderWidth": "2px",
              "height": "200px",
              "width": "400px",
            }
          }
        >
          <p>
            Try dropping some files here, or click to select files to upload.
          </p>
        </Dropzone>
      </div>
      <div
        id="combineButton"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <aside>
          <h2>
            Selected files
          </h2>
          <div />
        </aside>
        <input
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#2F4F4F",
              "border": "1px solid #566963",
              "borderRadius": "5px",
              "color": "#ffffff",
              "cursor": "pointer",
              "display": "inline-block",
              "fontSize": "15px",
              "fontWeight": "bold",
              "height": "50px",
              "margin": "10px",
              "textDecoration": "none",
              "textShadow": "0px-1px0px #2b665e",
            }
          }
          type="button"
          value="Combine Packages"
        />
      </div>
    </section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <AddPackage />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": AddPackage {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "files": Array [],
          "packageXML": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <section>
          <div
            className="dropzone"
          >
            <Dropzone
              accept=".xml"
              disableClick={false}
              disablePreview={false}
              disabled={false}
              maxSize={Infinity}
              minSize={0}
              multiple={true}
              onDrop={[Function]}
              preventDropOnDocument={true}
              style={
                Object {
                  "borderColor": "#666666",
                  "borderRadius": "5px",
                  "borderStyle": "dashed",
                  "borderWidth": "2px",
                  "height": "200px",
                  "width": "400px",
                }
              }
            >
              <p>
                Try dropping some files here, or click to select files to upload.
              </p>
            </Dropzone>
          </div>
          <div
            id="combineButton"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <aside>
              <h2>
                Selected files
              </h2>
              <div />
            </aside>
            <input
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#2F4F4F",
                  "border": "1px solid #566963",
                  "borderRadius": "5px",
                  "color": "#ffffff",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "fontSize": "15px",
                  "fontWeight": "bold",
                  "height": "50px",
                  "margin": "10px",
                  "textDecoration": "none",
                  "textShadow": "0px-1px0px #2b665e",
                }
              }
              type="button"
              value="Combine Packages"
            />
          </div>
        </section>,
        "_debugID": 4,
        "_renderedOutput": <section>
          <div
            className="dropzone"
          >
            <Dropzone
              accept=".xml"
              disableClick={false}
              disablePreview={false}
              disabled={false}
              maxSize={Infinity}
              minSize={0}
              multiple={true}
              onDrop={[Function]}
              preventDropOnDocument={true}
              style={
                Object {
                  "borderColor": "#666666",
                  "borderRadius": "5px",
                  "borderStyle": "dashed",
                  "borderWidth": "2px",
                  "height": "200px",
                  "width": "400px",
                }
              }
            >
              <p>
                Try dropping some files here, or click to select files to upload.
              </p>
            </Dropzone>
          </div>
          <div
            id="combineButton"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <aside>
              <h2>
                Selected files
              </h2>
              <div />
            </aside>
            <input
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#2F4F4F",
                  "border": "1px solid #566963",
                  "borderRadius": "5px",
                  "color": "#ffffff",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "fontSize": "15px",
                  "fontWeight": "bold",
                  "height": "50px",
                  "margin": "10px",
                  "textDecoration": "none",
                  "textShadow": "0px-1px0px #2b665e",
                }
              }
              type="button"
              value="Combine Packages"
            />
          </div>
        </section>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <AddPackage />,
}
`;

exports[`Creating the component does not fail should match its empty snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <section>
    <div
      className="dropzone"
    >
      <Dropzone
        accept=".xml"
        disableClick={false}
        disablePreview={false}
        disabled={false}
        maxSize={Infinity}
        minSize={0}
        multiple={true}
        onDrop={[Function]}
        preventDropOnDocument={true}
        style={
          Object {
            "borderColor": "#666666",
            "borderRadius": "5px",
            "borderStyle": "dashed",
            "borderWidth": "2px",
            "height": "200px",
            "width": "400px",
          }
        }
      >
        <p>
          Try dropping some files here, or click to select files to upload.
        </p>
      </Dropzone>
    </div>
    <div
      id="combineButton"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <aside>
        <h2>
          Selected files
        </h2>
        <div />
      </aside>
      <input
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#2F4F4F",
            "border": "1px solid #566963",
            "borderRadius": "5px",
            "color": "#ffffff",
            "cursor": "pointer",
            "display": "inline-block",
            "fontSize": "15px",
            "fontWeight": "bold",
            "height": "50px",
            "margin": "10px",
            "textDecoration": "none",
            "textShadow": "0px-1px0px #2b665e",
          }
        }
        type="button"
        value="Combine Packages"
      />
    </div>
  </section>,
  "nodes": Array [
    <section>
      <div
        className="dropzone"
      >
        <Dropzone
          accept=".xml"
          disableClick={false}
          disablePreview={false}
          disabled={false}
          maxSize={Infinity}
          minSize={0}
          multiple={true}
          onDrop={[Function]}
          preventDropOnDocument={true}
          style={
            Object {
              "borderColor": "#666666",
              "borderRadius": "5px",
              "borderStyle": "dashed",
              "borderWidth": "2px",
              "height": "200px",
              "width": "400px",
            }
          }
        >
          <p>
            Try dropping some files here, or click to select files to upload.
          </p>
        </Dropzone>
      </div>
      <div
        id="combineButton"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <aside>
          <h2>
            Selected files
          </h2>
          <div />
        </aside>
        <input
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#2F4F4F",
              "border": "1px solid #566963",
              "borderRadius": "5px",
              "color": "#ffffff",
              "cursor": "pointer",
              "display": "inline-block",
              "fontSize": "15px",
              "fontWeight": "bold",
              "height": "50px",
              "margin": "10px",
              "textDecoration": "none",
              "textShadow": "0px-1px0px #2b665e",
            }
          }
          type="button"
          value="Combine Packages"
        />
      </div>
    </section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <AddPackage />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": AddPackage {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "files": Array [],
          "packageXML": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <section>
          <div
            className="dropzone"
          >
            <Dropzone
              accept=".xml"
              disableClick={false}
              disablePreview={false}
              disabled={false}
              maxSize={Infinity}
              minSize={0}
              multiple={true}
              onDrop={[Function]}
              preventDropOnDocument={true}
              style={
                Object {
                  "borderColor": "#666666",
                  "borderRadius": "5px",
                  "borderStyle": "dashed",
                  "borderWidth": "2px",
                  "height": "200px",
                  "width": "400px",
                }
              }
            >
              <p>
                Try dropping some files here, or click to select files to upload.
              </p>
            </Dropzone>
          </div>
          <div
            id="combineButton"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <aside>
              <h2>
                Selected files
              </h2>
              <div />
            </aside>
            <input
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#2F4F4F",
                  "border": "1px solid #566963",
                  "borderRadius": "5px",
                  "color": "#ffffff",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "fontSize": "15px",
                  "fontWeight": "bold",
                  "height": "50px",
                  "margin": "10px",
                  "textDecoration": "none",
                  "textShadow": "0px-1px0px #2b665e",
                }
              }
              type="button"
              value="Combine Packages"
            />
          </div>
        </section>,
        "_debugID": 2,
        "_renderedOutput": <section>
          <div
            className="dropzone"
          >
            <Dropzone
              accept=".xml"
              disableClick={false}
              disablePreview={false}
              disabled={false}
              maxSize={Infinity}
              minSize={0}
              multiple={true}
              onDrop={[Function]}
              preventDropOnDocument={true}
              style={
                Object {
                  "borderColor": "#666666",
                  "borderRadius": "5px",
                  "borderStyle": "dashed",
                  "borderWidth": "2px",
                  "height": "200px",
                  "width": "400px",
                }
              }
            >
              <p>
                Try dropping some files here, or click to select files to upload.
              </p>
            </Dropzone>
          </div>
          <div
            id="combineButton"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <aside>
              <h2>
                Selected files
              </h2>
              <div />
            </aside>
            <input
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#2F4F4F",
                  "border": "1px solid #566963",
                  "borderRadius": "5px",
                  "color": "#ffffff",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "fontSize": "15px",
                  "fontWeight": "bold",
                  "height": "50px",
                  "margin": "10px",
                  "textDecoration": "none",
                  "textShadow": "0px-1px0px #2b665e",
                }
              }
              type="button"
              value="Combine Packages"
            />
          </div>
        </section>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <AddPackage />,
}
`;
